### PR TITLE
Remove vSphere VCH extension in vic-machine delete

### PR DESCRIPTION
**Pull request by gigawhitlocks**
Adds functionality to clean out the VCH extension when `vic-machine delete` is run.
Pretty minor/straightforward addition.

### CODE REVIEW

**Comment by gigawhitlocks** [comment]
@emlin you might want to look at this cause it's touching your code (but only what's already on master)

**Comment by andrewtchin** [comment]
lgtm

File: cmd/vic-machine/delete/delete.go [hunk 1]
```diff
@@ -143,6 +143,16 @@ func (d *Uninstall) Run(cli *cli.Context) error {
 		return err
 	}
 
+	if validator.IsVC() {
+		log.Infoln("Removing VCH vSphere extension")
+		if err = executor.GenerateExtensionName(vchConfig); err != nil {
+			return err
```
Comment (emlin) [inline cmd/vic-machine/delete/delete.go]: We should continue uninstallation even if the extension unregistration failed. You can simply warning the error, and continue

Comment (gigawhitlocks) [inline cmd/vic-machine/delete/delete.go]: Oh ok makes sense

File: cmd/vic-machine/delete/delete.go [hunk 2]
```diff
@@ -146,10 +146,10 @@ func (d *Uninstall) Run(cli *cli.Context) error {
 	if validator.IsVC() {
 		log.Infoln("Removing VCH vSphere extension")
 		if err = executor.GenerateExtensionName(vchConfig); err != nil {
-			return err
+			log.Errorf("Wasn't able to get extension name during VCH deletion. Failed with error: %s", err)
```
Comment (emlin) [inline cmd/vic-machine/delete/delete.go]: what about using warn here?

**Comment by emlin** [comment]
lgtm

**Comment by gigawhitlocks** [comment]
lgtm

**Pull request merged**